### PR TITLE
fix(curriculum): adds code examples for shift method for RPG Game

### LINKED ISSUE #53308: Add code example for shift method in RPG game
### Describe the Issue

When the `shift` method is introduced in the RPG game, a few people are using the incorrect syntax as evidenced in these forum posts
https://forum.freecodecamp.org/search?expanded=true&q=105%20%23javascript

They are trying to do something like this
```js
inventory.shift(sellWeapon);
```
The only other array method they learned earlier was the `push` method and so that is probably why they think they need to pass in a value for the `shift` method.

This confusion can be fixed by adding a small code example

Here is the updated description 

# --description--

The `shift()` method on an array removes the first element in the array and returns it. 

```js
const numbers = [1, 2, 3];
const firstNumber = numbers.shift(); // returns 1
```

Use the `shift()` method to take the first element from the `inventory` array and assign it to your `currentWeapon` variable.

### Affected Page

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-basic-javascript-by-building-a-role-playing-game/step-105

### Screenshots

_No response_

### System

- Device: [e.g. iPhone 6, Laptop]
- OS: [e.g. iOS 14, Windows 10, Ubuntu 20.04]
- Browser: [e.g. Chrome, Safari]
- Version: [e.g. 22]

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by tsirianni**
Checklist:

- [X] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [X] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [X] I have tested these changes either locally on my machine, or GitPod.

Closes #53308

### CODE REVIEW

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62aa21971e3b743844849985.md [hunk 1]
```diff
@@ -7,23 +7,23 @@ dashedName: step-151
 
 # --description--
 
-The player should hit if either `Math.random() > .2` **or** if the player's health is less than 20.
+The player should hit if either `Math.random() > .2` **or** if the player's health is less than `20``.
```
Comment (jdwilkin4) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62aa21971e3b743844849985.md:10]: All of the these changes look to be unrelated to the step about the shift method.
Plus, we already have a PR to address this issue.

Please revert all changes for this file back to its original state 👍

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62aa21971e3b743844849985.md
Comment (gikf) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62aa21971e3b743844849985.md:1]: Looks like this file has changes unrelated to the PR and linked issue. They need to be removed from the PR.

Comment (tsirianni) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62aa21971e3b743844849985.md:1]: Ok, no problems!

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62a8d539dc11cb42b5dd7ec8.md [hunk 2]
```diff
@@ -7,7 +7,14 @@ dashedName: step-104
...
 
-The `shift()` method on an array removes the first element in the array and returns it. Use this method to take the first element from the `inventory` array and assign it to your `currentWeapon` variable.
+The `shift()` method on an array removes the first element in the array and returns it.
+
+```js
+const numbers = [1, 2, 3];
+const firstNumber = numbers.shift(); // returns 1
+```
+
+Use this method to take the first element from the `inventory` array and assign it to your `currentWeapon` variable.
```
Comment (jdwilkin4) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62a8d539dc11cb42b5dd7ec8.md:17]: Let'e mention the method name again just like the issue mentioned 👍 
```suggestion
Use the `shift()` method to take the first element from the `inventory` array and assign it to your `currentWeapon` variable.
```

**Review by jdwilkin4 — CHANGES_REQUESTED** [review]
Thank you for your contribution!

Please review the requested changes. 👍

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62a8d539dc11cb42b5dd7ec8.md (hunk 2 above)
Comment (tsirianni) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62a8d539dc11cb42b5dd7ec8.md]: Done :)

**Review by jdwilkin4 — APPROVED** [review]
Congrats on your first contribution to our repository. 🎉
We look forward to future contributions. 👍

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62a8d539dc11cb42b5dd7ec8.md [hunk 3]
```diff
@@ -7,7 +7,14 @@ dashedName: step-104
 
 # --description--
 
-The `shift()` method on an array removes the first element in the array and returns it. Use this method to take the first element from the `inventory` array and assign it to your `currentWeapon` variable.
+The `shift()` method on an array removes the first element in the array and returns it.
+
+```js
+const numbers = [1, 2, 3];
+const firstNumber = numbers.shift(); // returns 1
```
Comment (gikf) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62a8d539dc11cb42b5dd7ec8.md:14]: Hmm, I'm not sure about comment here. It might be better to add another sentence with something like:

> After executing code below/above, the `firstNumber` will be equal to `1`.

@jdwilkin4 what do you think? This also would limit issues with translating, altought there isn't really that much to translate in `returns 1`.

Comment (jdwilkin4) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62a8d539dc11cb42b5dd7ec8.md:14]: I think either option works. 
We can either have another sentence or leave the comment.
We have had comments in other projects like this one 

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-basic-string-and-array-methods-by-building-a-music-player/step-9

Comment (gikf) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62a8d539dc11cb42b5dd7ec8.md:14]: Ah, then this is fine with me.

**Review by gikf — APPROVED** [review]
